### PR TITLE
Correctly select the AOM Profile for other models than OpenEHR

### PR DESCRIPTION
**Pull request by pieterbos**
The AOM profile was incorrectly selected just on publisher. Change this to selecting on BMM Schema id - with a fallback to publisher.

### CODE REVIEW

**Comment by pieterbos** [comment]
Maybe fallback needs to be in a separate loop instead?

File: aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java [hunk 1]
```diff
@@ -121,12 +121,11 @@ public void selectModel(String rmPublisher, String rmPackage, String rmRelease)
             selectedBmmModel = validationResult == null ? null : validationResult.getModel();
         }
 
-        for(AomProfile profile:aomProfiles.getProfiles()) {
-            if(profile.getProfileName().equalsIgnoreCase(rmPublisher)) {
-                this.selectedAomProfile = profile;
-                break;
-            }
+        AomProfile profile = getAomProfileWithSchemaId(selectedBmmModel);
```
Comment (J3173) [inline aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]: I think directly setting `this.selectedAomProfile`, without the local variable `profile` is clearer.

File: aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java [hunk 2]
```diff
@@ -135,6 +134,26 @@ public void selectModel(String rmPublisher, String rmPackage, String rmRelease)
 
     }
 
+    private AomProfile getAomProfileWithSchemaId(BmmModel selectedBmmModel) {
+        if(selectedBmmModel != null) {
+            for (AomProfile profile : aomProfiles.getProfiles()) {
+                if (profile.getRmSchemaPattern().stream().anyMatch(pat -> selectedBmmModel.getSchemaId().matches(pat))) {
+                    return this.selectedAomProfile = profile;
```
Comment (J3173) [inline aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]: You already set `this.selectedAomProfile` in the calling method.
```suggestion
                    return profile;
```

File: aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java [hunk 3]
```diff
@@ -135,6 +134,26 @@ public void selectModel(String rmPublisher, String rmPackage, String rmRelease)
...
+                }
+            }
+        }
+        return null;
+    }
+
+    private AomProfile getAomProfileOnPublisher(String rmPublisher) {
+        for(AomProfile profile:aomProfiles.getProfiles()) {
+           if(profile.getProfileName().equalsIgnoreCase(rmPublisher)) {
+                return this.selectedAomProfile = profile;
```
Comment (J3173) [inline aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]:
```suggestion
                return profile;
```

File: aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java (hunk 1 above)
Comment (pieterbos) [reply aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]: Yes, but then we need to do a ```this.selectedAomProfile = null;``` as a first line. I could instead do:

```
this.selectedAomProfile = getAomProfile()
```

Then add a new method that does the current logic :)

File: aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java (hunk 2 above)
Comment (pieterbos) [reply aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]: oops, that was left over from earlier code. thanks

File: aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java (hunk 1 above)
Comment (J3173) [reply aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]: I was thinking of:
```java
        this.selectedAomProfile = getAomProfileWithSchemaId(selectedBmmModel);
        if(this.selectedAomProfile == null) {
            this.selectedAomProfile = getAomProfileOnPublisher(rmPublisher);
        }
```
I think this should work.

Comment (pieterbos) [reply aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java]: Oh of course. done :)

**Pull request merged**